### PR TITLE
Comments areas list search

### PR DESCRIPTION
**Pull request by Bopzor**

### CODE REVIEW

File: packages/app-core/src/modules/commentsArea/usecases/fetchCommentsAreas/fetchCommentsAreas.ts [hunk 1]
```diff
@@ -1,17 +1,23 @@
...
-  if (selectTotalCommentsAreas(getState()) > 0) {
-    return;
-  }
+import { selectCommentsAreasSearchQuery } from '../../selectors';
 
+export const fetchCommentsAreas = createThunk(async ({ getState, dispatch, commentsAreaGateway }, query?: string) => {
   dispatch(setFetchingCommentsAreas(true));
 
-  const commentsAreasDtos = await commentsAreaGateway.fetchCommentsAreas();
+  const search = query ?? selectCommentsAreasSearchQuery(getState());
```
Comment (nilscox) [inline packages/app-core/src/modules/commentsArea/usecases/fetchCommentsAreas/fetchCommentsAreas.ts:11]: What about removing this use case's `query` parameter, and always use the value from the store?

File: packages/app-core/src/modules/commentsArea/usecases/searchCommentsAreas/searchCommentsAreas.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+const SEARCH_DEBOUNCE_TIMEOUT = 500;
+
+let searchTimeout: number | undefined;
+
+export const searchCommentsAreas = createThunk(async ({ dispatch, getState, timerGateway }, query: string) => {
+  if (searchTimeout) {
+    timerGateway.clearTimeout(searchTimeout);
+  }
+
+  if (!selectIsFetchingCommentsAreas(getState())) {
```
Comment (nilscox) [inline packages/app-core/src/modules/commentsArea/usecases/searchCommentsAreas/searchCommentsAreas.ts:15]: Let's invert this condition and return if the comments areas list is already being fetched.

File: packages/app-core/src/modules/commentsArea/usecases/searchCommentsAreas/searchCommentsAreas.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    if (query === '') {
+      await dispatch(clearSearch());
+    } else {
+      await dispatch(setSearch(query));
+    }
+  }, SEARCH_DEBOUNCE_TIMEOUT);
+});
+
+const setSearch = createThunk(async ({ dispatch }, query: string) => {
+  await dispatch(fetchCommentsAreas(query));
```
Comment (nilscox) [inline packages/app-core/src/modules/commentsArea/usecases/searchCommentsAreas/searchCommentsAreas.ts:29]: We'll need to set the search query first according to the first comment.

File: packages/app-infra/src/views/CommentsAreasListView/CommentsAreasListView.tsx [hunk 4]
```diff
@@ -13,17 +18,24 @@ import { List } from '~/components/layout/List/List';
...
 import { useHandleAuthenticationTokens } from './useHandleAuthenticationTokens';
+import useUpdateEffect from '~/hooks/useUpdateEffect';
 
 export const CommentsAreasListView: React.FC = () => {
   const dispatch = useDispatch();
 
   const commentsAreas = useAppSelector(selectCommentsAreas);
   const loading = useAppSelector(selectIsFetchingCommentsAreas);
 
+  const [search, setSearch] = useState('');
```
Comment (nilscox) [inline packages/app-infra/src/views/CommentsAreasListView/CommentsAreasListView.tsx:29]: We can avoid this state by using the value from the store directly.

**Review by nilscox — CHANGES_REQUESTED** [review]
Great work! :pray:

File: packages/app-core/src/modules/commentsArea/usecases/fetchCommentsAreas/fetchCommentsAreas.ts (hunk 1 above)
Comment (Bopzor) [reply packages/app-core/src/modules/commentsArea/usecases/fetchCommentsAreas/fetchCommentsAreas.ts:11]: I think it's a good idea. No need of source duplication :)

**Review by nilscox — APPROVED** [review]

File: packages/app-core/src/modules/commentsArea/usecases/searchCommentsAreas/searchCommentsAreas.ts (hunk 2 above)
Comment (nilscox) [reply packages/app-core/src/modules/commentsArea/usecases/searchCommentsAreas/searchCommentsAreas.ts:15]: Not possible due to the debounce being handled by the domain.

**Pull request merged**